crates/bevy_reflect/src/func/info.rs

```diff
@@ -0,0 +1,108 @@
+use crate::func::args::{ArgInfo, GetOwnership, Ownership};
+use crate::TypePath;
+use alloc::borrow::Cow;
+
+/// Type information for a [`DynamicFunction`].
+///
+/// [`DynamicFunction`]: crate::func::DynamicFunction
+#[derive(Debug, Clone)]
+pub struct FunctionInfo {
+    name: Option<Cow<'static, str>>,
+    args: Vec<ArgInfo>,
+    return_info: ReturnInfo,
+}
+
+impl FunctionInfo {
+    /// Create a new [`FunctionInfo`].
+    pub fn new() -> Self {
+        Self {
+            name: None,
+            args: Vec::new(),
+            return_info: ReturnInfo::new::<()>(),
+        }
+    }
+
+    /// Set the name of the function.
+    pub fn with_name(mut self, name: impl Into<Cow<'static, str>>) -> Self {
+        self.name = Some(name.into());
+        self
+    }
+
+    /// Set the arguments of the function.
+    ///
+    /// It is very important that the arguments match the intended function signature,
```

Comment:
What are the consequences of failure here? UB? Failures?

Reply:
Not UB, just an error returned. I'll add a comment!